### DISCUSSION TITLE
Matter user experience

### DISCUSSION THREAD

**Discussion by jpadie** [arendst/Tasmota#19407] [Q&A]
i'm testing out the matter implementation on ESP32 and alexa.  I wanted to check the intended UX with the community.

1. turn "enable matter on". wait for reboot
2. configure matter local sensors/devices.  Four appear on my implementation (four relays). but I only want two.
3. so delete (the fire icon) the first two.  configure the parameters of the remaining two to correspond to the relay index (3 and 4 in my case).  Add meaningful names to each local sensor.
4. save 
5. go through device discovery and flash the QR code in the alexa app. 
6. after a (longish) time the alexa app recognises only one matter device and always names it "Tasmota".  i.e. it ignores the name given in the matter config page (and it also ignores the friendly name, by the by).  

Is it intended behaviour that only one device at a time should be found by a matter hub?  and is it intended behaviour that this is always with the name "Tasmota"?  

am using an OTA build from today of ver 13.1.0.

**Comment by sfromis** [comment]
I do not know much of details, but I've seen that "Alexa currently supports only one endpoint". And I'd try if configuring the "Friendly name" helps....

**Comment by jpadie** [comment]
thanks.   friendly names are not honoured by the connection either as per my initial post. 

i've not found the technical limitation that you reference in the Alexa dev documentation.  nothing specific but the inference from the dev material is the opposite.  

i'd be interested in Stephan's experience here, as if I recall he wrote the code-base for Matter on tasmota and he might have some in-house insights into Alexa.

**Reply by sfromis** [reply]
"Alexa currently supports only one endpoint" was a statement from Stephan.

**Reply by jpadie** [reply]
thanks @sfromis.  

i wonder whether this is still a limitation and where it comes from.  it's not seemingly a limitation within the Alexa SDK (but I have not dug super deep yet).

**Reply by s-hadinger** [reply]
The friendly name are only passed when in "bridge" mode - this is as per Matter protocol. Up to now Alexa didn't support bridge mode, but I was just told that it does support it now.

I need to do further tests.
